.semaphore/semaphore.yml

```diff
@@ -0,0 +1,57 @@
+version: v1.0
+name: confluent-docker-utils
+agent:
+  machine:
+    type: s1-prod-ubuntu20-04-amd64-1
+execution_time_limit:
+  minutes: 30
+global_job_config:
+  prologue:
+    commands:
+      - checkout
+      - sem-version python 3.8
+      - pip install tox
+      - COMMIT_MESSAGE_PREFIX="Publish version"
+      - LAST_COMMIT_MESSAGE=$(git log -1 --oneline --format="%ae %s")
+      - >- 
+        if [[ "$LAST_COMMIT_MESSAGE" =~ "semaphore@confluent.io $COMMIT_MESSAGE_PREFIX" ]]; then
+          echo "Skipping publish for changes from a Semaphore Version Bump"
+          exit 0
+        fi
+blocks:
+  - name: "Tests"
+    dependencies: []
+    task:
+      jobs:
+        - name: "Tests"
+          commands:
+            - tox
+  - name: "Release"
+    dependencies: ["Tests"]
+    run:
+      when: "branch = 'master'"
+    task:
+      jobs:
+        - name: "Release"
+          commands:
+            - VIRTUAL_ENV=/tmp/publish
+            - python3 -m venv $VIRTUAL_ENV
+            - source $VIRTUAL_ENV/bin/activate
+            - pip install wheel==0.33.1 bumpversion==0.5.3 Mako==1.0.8 MarkupSafe==1.1.1
```

Comment:
FYI Bazel builds bump version like https://github.com/confluentinc/confluent-bazel-rules/blob/master/python/scripts/bump_version.sh. The implementation was copied from the Jenkins logic and then tailored to Semaphore. You can't use the Bazel logic directly, but it would be good to compare and update your logic as necessary. I am not too concerned with implementation details as long as it works, but since you can only test this in production, it might be worth making some changes.

Reply:
the implementation here was also taken from Jenkins logic. from the looks of it it's very similar (we both prob copied the same script). There's some bazel specific changes the bazel implementation has, otherwise v similar already.